### DISCUSSION TITLE
🔢 Seastar Dataset

### DISCUSSION THREAD

**Discussion by JoelMathewC** [bfGraph/STGraph#33] [Benchmarking]
Collection of datasets provided by Seastar to be used for GNN model training

**Comment by JoelMathewC** [comment]
We've decided to go ahead with 3 medium size datasets and 2 small size datasets for dynamic graph benchmarking

The small size datasets are 
1. [EnglandCovid](https://pytorch-geometric-temporal.readthedocs.io/en/latest/_modules/torch_geometric_temporal/dataset/encovid.html#EnglandCovidDatasetLoader)
2. [Twitter Tennis](https://pytorch-geometric-temporal.readthedocs.io/en/latest/_modules/torch_geometric_temporal/dataset/twitter_tennis.html#TwitterTennisDatasetLoader)

The medium size datasets are available [snap-stanford](http://snap.stanford.edu/jodie/) and [twitter-tgn](https://github.com/twitter-research/tgn)
1. Wikipedia
2. Reddit
3. Twitter

**Comment by JoelMathewC** [comment]
## Wikipedia Dataset

1. Public dataset from one month of edits made on Wikipedia pages
2. Bipartite graph (8227 user nodes and 1000 pages nodes)
3. Feature vectors are LIWC (Linguistic Inquiry and word count) -> feature vector representing the text
4. Nodes are users and pages, edges indicate interaction of users with pages
5. The dataset is used to identify when a user is likely to turn malicious on Wikipedia. This ensures integrity
6. The dataset is of format _user, item, timestamp, state label, comma-separated array of features_
7. state label is 0 for non-malicious and 1 for malicious (They have collected this data by checking which user got banned)
8. **Aim:** Predict each users next interaction and change in state label of user [Node classification, Edge Prediction]
9. Initialize all users and page nodes with the same node embedding as shown [here](https://github.com/claws-lab/jodie/blob/177409bf8d173d4200547e2d5bf2dd0f701ebb77/jodie.py#L72)
10. **Loading the data:** We can follow this [approach](https://github.com/claws-lab/jodie/blob/177409bf8d173d4200547e2d5bf2dd0f701ebb77/library_data.py#L19)
11. **Batching Timespan:** JODIE uses a T-Batch algorithm we can do something similar as shown [here](https://github.com/claws-lab/jodie/blob/master/tbatch.py)

**Comment by JoelMathewC** [comment]
## Stanford SNAP

The category is defined as networks where edges have timestamps.

**Comment by JoelMathewC** [comment]
## OGB

We can consider using the node predition datasets available [here](https://ogb.stanford.edu/docs/nodeprop/) but these are not really temporal in nature, since at every new timestamp the old feature vectors do not change. Its just that new nodes get added and edges associated to those new nodes get added.

**Comment by JoelMathewC** [comment]
## On the Hunt

We are on the lookout for models that can be used for the edge prediction task. There are three candidates at the moment
1. GC-LSTM (cited 141 times)
2. EvolveGCN (cited 625 times)
3. DySAT (cited 307 times)

We need to find which one we can use and implement in Seastar.

**Comment by JoelMathewC** [comment]
We did end up using a total of 10 datasets (5 static and 5 DTDGs) with the TGCN model for benchmarking. Closing this out.
